Assets/PatchKit Patcher/Scripts/AppData/Local/Pack1Unarchiver.cs

```diff
@@ -215,45 +246,47 @@ private void UnpackRegularFile(Pack1Meta.FileEntry file, Action<double> onProgre
             DebugLogger.Log("File " + file.Name + " unpacked successfully!");
         }
 
+        private Stream CreateXzDecompressor(Stream source)
+        {
+            if (source.CanSeek)
+            {
+                return new XZStream(source);
+            }
+            else
+            {
+                return new XZStream(new PositionAwareStream(source));
+            }
+        }
+
+        private Stream CreateGzipDecompressor(Stream source)
+        {
+            return new GZipStream(source, CompressionMode.Decompress);
+        }
+
         private void ExtractFileFromStream(
-            Stream sourceStream,
+            LimitedStream sourceStream,
             Stream targetStream,
-            Pack1Meta.FileEntry file,
+            long fileSize,
             ICryptoTransform decryptor,
+            DecompressorCreator createDecompressor,
             Action<double> onProgress,
             CancellationToken cancellationToken)
         {
             using (var cryptoStream = new CryptoStream(sourceStream, decryptor, CryptoStreamMode.Read))
             {
-                using (var gzipStream = new GZipStream(cryptoStream, Ionic.Zlib.CompressionMode.Decompress))
+                using (var decompressionStream = createDecompressor(cryptoStream))
                 {
-                    try
-                    {
-                        long bytesProcessed = 0;
-                        const int bufferSize = 128 * 1024;
-                        var buffer = new byte[bufferSize];
-                        int count;
-                        while ((count = gzipStream.Read(buffer, 0, bufferSize)) != 0)
-                        {
-                            targetStream.Write(buffer, 0, count);
-                            bytesProcessed += count;
-                            onProgress(bytesProcessed / (double) file.Size.Value);
-                        }
-                    }
-                    catch (Exception e)
-                    {
-                        DebugLogger.LogException(e);
-
-                        var logManager = PatcherLogManager.Instance;
```

Comment:
Catching and reporting uncompression exception has been removed from here. Is it intentional?

Reply:
Intentional at the time but completely misguided, thanks for noticing.

Reply:
So it will be brought back right?

Reply:
Yup, I'll get to it today